### PR TITLE
requireDollarBeforejQueryAssignment: Add ignoreProperties rule value

### LINKED ISSUE #1266: jQuery identifiers must start with a $ - failing for object properties
Using JSCS/gulp on a project which assigns a jquery object to an object property triggers the `jQuery identifiers must start with a $` error. However, the property is part of a configuration object for another javascript in bower, and outside our control.

```
jQuery identifiers must start with a $ at assets/js/controllers/project/fields.js :
    24 |        $scope.initFormBuilder = function() {
    25 |            $scope.formbuilder = new Formbuilder({
    26 |                targets: $('.formbuilder'),
------------------------^
```

Assigning it to a variable first solves the issue, but is unnecessary in this specific case.

Using gulp-jscs 1.4.0 on Mac OS X 10.10.2.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/11092674-jquery-identifiers-must-start-with-a-failing-for-object-properties?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by zxqfox**
Fixes #1266

### CODE REVIEW

**Comment by zxqfox** [comment]
Feels like what we need. Code is a little bit complicated but looks like we need to left just variables check with `ignoreProperties`. Am I right?

/cc @viveleroi

**Comment by TheSavior** [comment]
Frankly, I'm surprised this works. Mostly because I would think we shouldn't return if ignoreProperties is true, but only if we are inside of an object *and* ignoreProperties is true. 

However, I couldn't find a test that previously had an error, that should still have an error with this relaxing modification, so maybe it is okay.

File: lib/rules/require-dollar-before-jquery-assignment.js [hunk 1]
```diff
@@ -53,6 +79,10 @@ module.exports.prototype = {
                 left = token.id;
                 varName = left.name;
                 right = token.init;
+
+            } else if (ignoreProperties) {
+                return;
+
```
Comment (zxqfox) [inline lib/rules/require-dollar-before-jquery-assignment.js]: @markelog @mikesherov What our code style said about padding lines about `else if`s?

File: lib/rules/require-dollar-before-jquery-assignment.js [hunk 2]
```diff
@@ -61,6 +91,7 @@ module.exports.prototype = {
 
                 varName = left.name || left.property.name;
                 right = token.right;
+
```
Comment (zxqfox) [inline lib/rules/require-dollar-before-jquery-assignment.js]: random whitespace.

**Comment by markelog** [comment]
Inconsistency in rule values is frustrating though

**Comment by zxqfox** [comment]
@markelog Let's fix it! I've used `ignoreProperties` because it was suggested here https://github.com/jscs-dev/node-jscs/issues/1266#issuecomment-92838858

Any ideas?

**Comment by markelog** [comment]
`allExcept` comes to mind, but this would not save us, since we already use `ignoreProperties` in another rule.

2.0 should save us all

**Comment by zxqfox** [comment]
@markelog Made an issue for that: #1303

**Pull request closed without merging**